server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -56,8 +56,7 @@ public DesiredBalance compute(
         final var routingNodes = routingAllocation.routingNodes();
         final var ignoredShards = new HashSet<>(desiredBalanceInput.ignoredShards());
         final var changes = routingAllocation.changes();
-        final var knownNodeIds = routingAllocation.nodes().stream().map(DiscoveryNode::getId).collect(Collectors.toSet());
-        final var unassignedPrimaries = new HashSet<ShardId>();
+        final var clusterInfoSimulator = new ClusterInfoSimulator(routingAllocation.clusterInfo());
```

Comment:
Could we enhance `DesiredBalanceComputerTests` to verify we're accounting for disk usage in all the right places?